apps/wing/src/commands/lsp.ts

```diff
@@ -3,25 +3,47 @@ import {
   InitializeParams,
   TextDocumentSyncKind,
   InitializeResult,
-  CompletionItem,
-  DocumentSymbol,
-  Hover,
-  LocationLink,
 } from "vscode-languageserver/node";
 
 import * as wingCompiler from "../wingc";
 
 export async function run_server() {
-  const wingc = await wingCompiler.load({
+  let wingc = await wingCompiler.load({
     imports: {
       env: {
         send_notification,
       },
     },
   });
+  let badState = false;
 
-  let connection = createConnection(process.stdin, process.stdout);
+  const callWing = (func: wingCompiler.WingCompilerFunction, args: any): any | null => {
+    if (badState) {
+      return null;
+    }
+
+    try {
+      const result = wingCompiler.invoke(wingc, func, JSON.stringify(args));
+      if (typeof result === "number") {
+        if (result === 0) {
+          return null;
+        } else {
+          return result;
+        }
+      } else {
+        return JSON.parse(result);
+      }
+    } catch (e) {
+      connection.window.showErrorMessage(
```

Comment:
Maybe this is too "in your face". We might want to quietly log this and recover without bothering the user (this is under the assumption that some diagnostic was reported to the vscode extension with relevant span info (so the crash looks like any other syntax error and is visible to the user).

Reply:
Since the compiler crashed there are no diagnostics available. So it is a little jarring to not be told why lsp features are suddenly not working.

I added a compromise though: Instead of a message a new diagnostic error is added with the same message. Less "in your face" but still pretty visible.

Reply:
That's good, i expect this'll change soon to a real diag with my panic fixes.